### ISSUE TITLE
[a11y]: Icon button doesn't set `aria-pressed` with isSelected

### ISSUE THREAD

**Issue by warrenmblood** [carbon-design-system/carbon#23004]
### Package

@carbon/web-components

### Browser

Chrome

### Operating System

MacOS

### Package version

v2.61.0

### Automated testing tool and ruleset

VoiceOver

### Description

Currently when `isSelected` is `true`, the selected class and styles are applied to the button (icon-only Ghost variant). However, `aria-pressed` is not set, so screen readers won't announce the selected state. Verified the React button correctly sets `aria-pressed` so this is only a WC issue.

This issue is also a blocker for AI chat [#1963](https://github.com/carbon-design-system/carbon-ai-chat/issues/1963) @davidmenendez 

### Reproduction/example

https://web-components.carbondesignsystem.com/?path=/story/components-button--icon-button

### Steps to reproduce

- Go to Icon button [Story ](https://web-components.carbondesignsystem.com/?path=/story/components-button--icon-button)
- In controls, set isSelected to true and kind to ghost
- Inspect button in dev tools and see that `aria-pressed` is not set on `<button>`
- Turn on Voiceover or other screen reader and observe the selected state is not announced

### Suggested Severity

Critical (Severity 2) = User cannot complete task, and/or no workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
